src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomGeneratorSettingsFixer.java

```diff
@@ -0,0 +1,467 @@
+        newRoot.add("periodicGaussianOres", getPeriodicGaussianOres(oldRoot));
+        newRoot.add("expectedBaseHeight", getExpectedBaseHeight(oldRoot));
+        newRoot.add("expectedHeightVariation", getExpectedHeightVariation(oldRoot));
+        newRoot.add("actualHeight", getActualHeight(oldRoot));
+        newRoot.add("heightVariationFactor", getHeightVariationFactor(oldRoot));
+        newRoot.add("specialHeightVariationFactorBelowAverageY", getSpecialHeightVariationFactorBelowAverageY(oldRoot));
+        newRoot.add("heightVariationOffset", getHeightVariationOffset(oldRoot));
+        newRoot.add("heightFactor", getHeightFactor(oldRoot));
+        newRoot.add("heightOffset", getHeightOffset(oldRoot));
+        newRoot.add("depthNoiseFactor", getDepthNoiseFactor(oldRoot));
+        newRoot.add("depthNoiseOffset", getDepthNoiseOffset(oldRoot));
+        newRoot.add("depthNoiseFrequencyX", getDepthNoiseFrequencyX(oldRoot));
+        newRoot.add("depthNoiseFrequencyZ", getDepthNoiseFrequencyZ(oldRoot));
+        newRoot.add("depthNoiseOctaves", getDepthNoiseOctaves(oldRoot));
+        newRoot.add("selectorNoiseFactor", getSelectorNoiseFactor(oldRoot));
+        newRoot.add("selectorNoiseOffset", getSelectorNoiseOffset(oldRoot));
+        newRoot.add("selectorNoiseFrequencyX", getSelectorNoiseFrequencyX(oldRoot));
+        newRoot.add("selectorNoiseFrequencyY", getSelectorNoiseFrequencyY(oldRoot));
+        newRoot.add("selectorNoiseFrequencyZ", getSelectorNoiseFrequencyZ(oldRoot));
+        newRoot.add("selectorNoiseOctaves", getSelectorNoiseOctaves(oldRoot));
+        newRoot.add("lowNoiseFactor", getLowNoiseFactor(oldRoot));
+        newRoot.add("lowNoiseOffset", getLowNoiseOffset(oldRoot));
+        newRoot.add("lowNoiseFrequencyX", getLowNoiseFrequencyX(oldRoot));
+        newRoot.add("lowNoiseFrequencyY", getLowNoiseFrequencyY(oldRoot));
+        newRoot.add("lowNoiseFrequencyZ", getLowNoiseFrequencyZ(oldRoot));
+        newRoot.add("lowNoiseOctaves", getLowNoiseOctaves(oldRoot));
+        newRoot.add("highNoiseFactor", getHighNoiseFactor(oldRoot));
+        newRoot.add("highNoiseOffset", getHighNoiseOffset(oldRoot));
+        newRoot.add("highNoiseFrequencyX", getHighNoiseFrequencyX(oldRoot));
+        newRoot.add("highNoiseFrequencyY", getHighNoiseFrequencyY(oldRoot));
+        newRoot.add("highNoiseFrequencyZ", getHighNoiseFrequencyZ(oldRoot));
+        newRoot.add("highNoiseOctaves", getHighNoiseOctaves(oldRoot));
+        newRoot.add("cubeAreas", getCubeAreas(oldRoot));
+        newRoot.add("replacerConfig", getReplacerConfig(oldRoot));
+
+        String newGeneratorOptions = gson.toJson(newRoot).replaceAll("cubicchunks:", MODID + ":");
+        return newGeneratorOptions;
+    }
+
+    private static JsonElement getReplacerConfig(JsonObject json) {
+        Gson gson = CustomGeneratorSettings.gson(false);
+        return getOrDefault(json, "replacerConfig", gson.toJsonTree(BiomeBlockReplacerConfig.defaults()));
```

Comment:
Don't depend on any deserialized config objects. The way you did it, changing the replacer config class in the future (adding new options, changing the way replacers are implemented etc...) would affect this old datafixer. 

Old datafixers are supposed to be written once and then left there, unchanging, and never change what they do regardless of how MC or the current latest generator settings implementation changes.